app/controllers/users_controller.rb

```diff
@@ -176,11 +194,14 @@ def update_user(user_data)
       change_email(user_data)
     elsif user_data[:auto_follow_back]
       change_settings(user_data, "users.update.follow_settings_changed", "users.update.follow_settings_not_changed")
+    elsif user_data[:post_default_public]
+      change_post_default(user_data)
     elsif user_data[:color_theme]
       change_settings(user_data, "users.update.color_theme_changed", "users.update.color_theme_not_changed")
     else
       change_settings(user_data)
     end
+    edit
```

Comment:
The /contacts page uses an aspect_dropdown without being a stream IIRC, maybe you can have a look at how it's done there?